README.md

```diff
@@ -1,41 +1,112 @@
+
+The advantage to the above formalism is that it allows the zero forcing number to be computed using any of the standard algorithms for finding a path of smallest weight in a weighted directed graph.  In particular, in this implementation we use Dijkstra's Algorithm.
+
+### Capabilities and limitations.
+
+The software is able to compute the zero forcing number very efficiently for most simple graphs.  Some graphs, such as stars, represent a weakness for the algorithm and may produce longer running times.  In addition, memory usage can be prodigious for very large graphs.  Steps are planned to address both of these limitations in the future.  For now, the algorithm is very quick for most graphs.  For example, for the Paley graph on 101 vertices, the zero forcing number can be computed in just a few seconds.
 
 ---
 
-## Install from PyPI
-```bash
-sage -pip install -i https://test.pypi.org/simple/ zeroforcing
+
+### How to install:
+
+#### Option 1. Install through PyPI
+
+Type the line below into any cell in a SageMath Jupyter notebook:
+
+```
+%pip install --index-url https://test.pypi.org/simple/ zeroforcing==0.1.0
 ```
 
-## Install from source
-```bash
-sage -pip install .
+Alternatively, type the following at the command line in any terminal session:
+
+```
+sage -pip install --index-url https://test.pypi.org/simple/ zeroforcing==0.1.0
 ```
 
-## Test
-### Install the dependencies for testing
-```bash
-sage -pip install zeroforcing[test]
+
+The code should download from PyPI and build using Cython. (If running on macOS, during the process you may be prompted to install the command line developer tools.  This is because installation involves compiling Cython code.)
+
+#### Option 2: Install from source
+
+Either use `git` to clone the repository into the directory in which your SageMath Jupyter notebook resides, or follow these steps:
+
+* Download the `.zip` file of the project from [the releases page](https://github.com/alexhutman/ZeroForcingNumber/releases).
+* Move the `.zip` file into the same directory as your SageMath Jupyter notebook.
+
+Into any cell in the notebook, execute this command:
+
+```
+!unzip ZeroForcingNumber-master.zip
 ```
+Then execute this command:
 
-### Build the wavefront code for verification during tests
-```bash
-sage --python3 setup.py wavefront
 ```
+pip install .
+```
+
+#### Option 3: Run from Docker
+
+1. Obtain and install [Docker](https://www.docker.com/).
+2. Run `docker build -t zeroforcing .` in the directory this git repository is located in
+    * You can specify `--build-arg ZF_BUILD_ARGS="--debug"` before the `-t` flag to build in debug mode
+3. Run `docker run --rm -it zeroforcing`
+4. Follow the "How to use" section
 
-### Install the wavefront code we just built
-```bash
-sage -pip install .
+
+---
+
+### How to use:
```

Comment:
Same as [this comment](https://github.com/alexhutman/ZeroForcingNumber/pull/15/files#r1723713596)